mypyc/test-data/irbuild-lists.test

```diff
@@ -573,6 +573,228 @@ L0:
     r0 = CPySequence_Sort(a)
     return 1
 
+[case testListFromRange1]
+def fn() -> list[int]:
+    return list(range(3))
+[out]
+def fn():
+    r0, r1 :: object
+    r2 :: object[1]
+    r3 :: object_ptr
+    r4 :: object
+    r5 :: range
+    r6 :: list
+L0:
+    r0 = load_address PyRange_Type
+    r1 = object 3
+    r2 = [r1]
+    r3 = load_address r2
+    r4 = PyObject_Vectorcall(r0, r3, 1, 0)
+    keep_alive r1
+    r5 = cast(range, r4)
+    r6 = PySequence_List(r5)
+    return r6
+
+[case testListFromRange2]
+def fn() -> list[int]:
+    return list(range(1, 3))
+[out]
+def fn():
+    r0, r1, r2 :: object
+    r3 :: object[2]
+    r4 :: object_ptr
+    r5 :: object
+    r6 :: range
+    r7 :: list
+L0:
+    r0 = load_address PyRange_Type
+    r1 = object 1
+    r2 = object 3
+    r3 = [r1, r2]
+    r4 = load_address r3
+    r5 = PyObject_Vectorcall(r0, r4, 2, 0)
+    keep_alive r1, r2
+    r6 = cast(range, r5)
+    r7 = PySequence_List(r6)
+    return r7
+
+[case testListFromRange3]
+def fn() -> list[int]:
+    return list(range(1, 3, 2))
+[out]
+def fn():
+    r0, r1, r2, r3 :: object
+    r4 :: object[3]
+    r5 :: object_ptr
+    r6 :: object
+    r7 :: range
+    r8 :: list
+L0:
+    r0 = load_address PyRange_Type
+    r1 = object 1
+    r2 = object 3
+    r3 = object 2
+    r4 = [r1, r2, r3]
+    r5 = load_address r4
+    r6 = PyObject_Vectorcall(r0, r5, 3, 0)
+    keep_alive r1, r2, r3
+    r7 = cast(range, r6)
+    r8 = PySequence_List(r7)
+    return r8
+
+[case testListFromRange4]
+def fn() -> list[str]:
+    abc = tuple(range(3))
+    return list(str(i) for i in abc)
```

Comment:
This feels too specific for an irbuild test -- again, can you move this to be a run test?